dvc/repo/brancher.py

```diff
@@ -101,6 +102,8 @@ def brancher(  # noqa: E302
     finally:
         self.fs = saved_fs
         self.root_dir = saved_root
+        self.dvc_dir = saved_dvc_dir
+        self._reset()  # pylint: disable=protected-access
```

Comment:
Clunky and brancher should go away in favor of `switch`/etc, but this will do for now.